asab/library/providers/zookeeper.py

```
@@ -264,109 +262,179 @@ def _check_version_counter(self, version):
+		"""
+		Returns '/.personal/{tenant}/{cred}{path}' or None if tenant/cred are missing.
+		"""
+		assert path[:1] == '/'
+		if not tenant_id or not cred_id:
+			return None
+		return "{}/.personal/{}/{}{}".format(self.BasePath, tenant_id, cred_id, path).rstrip("/")
+
+	async def read(self, path: str) -> typing.Optional[typing.IO]:
 		if self.Zookeeper is None:
 			L.warning("Zookeeper Client has not been established (yet). Cannot read {}".format(path))
 			raise RuntimeError("Zookeeper Client has not been established (yet). Not ready.")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)